src/main/java/supercoder79/chunkpregen/PregenerationTask.java

```diff
@@ -158,26 +174,47 @@ private LongList collectChunks(int count) {
         LongList chunks = new LongArrayList(count);
 
         Iterator<ChunkPos> iterator = this.iterator;
-        for (int i = 0; i < count && iterator.hasNext(); i++) {
-            ChunkPos chunk = iterator.next();
-            chunks.add(ChunkPos.toLong(chunk.x + this.x, chunk.z + this.z));
+        for (int i = 0; i < count && iterator.hasNext();) {
+            ChunkPos chunkPosInLocalSpace = iterator.next();
+            if (Math.abs(chunkPosInLocalSpace.x) <= this.radius && Math.abs(chunkPosInLocalSpace.z) <= this.radius) {
```

Comment:
Nit: this seems like something that would make sense to include in `CoarseOnionIterator`'s logic? 🙂

Reply:
Hmm It seems I missed the real cause of the extra chunks getting generated was because of a missing check that x is less than radius. Adding the c check in CoarseOnionIterator seems to keep the % from going beyond 100% while reaching the end properly and generating the area as a perfect square. Huh